supportedBackends/jes/src/main/scala/cromwell/backend/impl/jes/JesCacheHitCopyingActor.scala

```diff
@@ -1,71 +1,104 @@
 import cromwell.core.{JobOutputs, PathCopier}
+import cromwell.services.metadata.MetadataService.PutMetadataAction
+import cromwell.services.metadata._
 import org.slf4j.LoggerFactory
+import wdl4s.values.WdlFile
 
 import scala.concurrent.Future
 
-object JesCacheHitCopyingActor {
-  val logger = LoggerFactory.getLogger("JesBackend")
-
-  def props(jobDescriptor: BackendJobDescriptor,
-            jesConfiguration: JesConfiguration,
-            initializationData: JesBackendInitializationData,
-            serviceRegistryActor: ActorRef): Props = {
-    Props(new JesCacheHitCopyingActor(jobDescriptor, jesConfiguration, initializationData, serviceRegistryActor))
-  }
-}
-
 case class JesCacheHitCopyingActor(override val jobDescriptor: BackendJobDescriptor,
                                    jesConfiguration: JesConfiguration,
                                    initializationData: JesBackendInitializationData,
                                    serviceRegistryActor: ActorRef) extends BackendCacheHitCopyingActor {
 
   override val configurationDescriptor = jesConfiguration.configurationDescriptor
+  private lazy val jesAttributes = jesConfiguration.jesAttributes
+  private lazy val metadataJobKey = {
+    val jobDescriptorKey: BackendJobDescriptorKey = jobDescriptor.key
+    MetadataJobKey(jobDescriptorKey.call.fullyQualifiedName, jobDescriptorKey.index, jobDescriptorKey.attempt)
+  }
 
-  override def copyCachedOutputs(cachedJobOutputs: JobOutputs): Future[BackendJobExecutionResponse] = {
-    val workflowPaths = initializationData.workflowPaths
-    val jesWorkflowPaths = workflowPaths.toJesCallPaths(jobDescriptor.key)
-    val callRootPath = jesWorkflowPaths.callRootPath
-    val callRootParent = callRootPath.getParent
-    val callRootRoot = callRootPath.getRoot
-
-    //renaming requires some tech talk
-    def renameCallSpecificFiles: List[String] = {
-      val jesFileNames = List(
-        jesWorkflowPaths.returnCodeFilename,
-        jesWorkflowPaths.stderrFilename,
-        jesWorkflowPaths.stdoutFilename,
-        jesWorkflowPaths.jesLogFilename
-      )
-      jesFileNames
+  override def copyCachedOutputs(seqOfSimpletons: Seq[WdlValueSimpleton], jobDetritusFiles: Map[String,String],
+                                 returnCode: Option[Int]): Future[BackendJobExecutionResponse] = {
+
+      val gcsFileSystem = initializationData.workflowPaths.gcsFileSystemWithUserAuth
+      val jesCallPaths = initializationData.workflowPaths.toJesCallPaths(jobDescriptor.key)
+      val destinationCallRootPath: Path = jesCallPaths.callRootPath
+      val getSourceCallRootPath = jobDetritusFiles.get(jesCallPaths.callRootPathKey) match {
+        case Some(srcPath) => gcsFileSystem.getPath(srcPath)
+        case None => val error = new RuntimeException(s"The call detritus files for source cache hit aren't found for call ${jobDescriptor.call.fullyQualifiedName}")
+      }
+
+    def cacheIfPossible = {
```

Comment:
For clarity, I'd really like to make this `def cacheIfPossible(): Future[BackendJobExecutionResponse]`
The `()` because this is a method with side-effects